samza-core/src/main/java/org/apache/samza/operators/impl/ControlMessageSender.java

```diff
@@ -48,20 +49,35 @@
   }
 
   void send(ControlMessage message, SystemStream systemStream, MessageCollector collector) {
-    Integer partitionCount = PARTITION_COUNT_CACHE.computeIfAbsent(systemStream, ss -> {
+    int partitionCount = getPartitionCount(systemStream);
+    int aggregatePartition = systemStream.getStream().hashCode() % partitionCount;
```

Comment:
please add a comment explaining what aggregatePartition is.

Reply:
Added the comment.